src/tools/tool_registry.py

```
@@ -96,6 +110,54 @@ def _register_conditional_tools(self):
                 "condition": lambda: os.getenv("USE_KNOWLEDGE_GRAPH", "false") == "true",
                 "condition_description": "Requires USE_KNOWLEDGE_GRAPH=true",
             },
+            "ingest_markdown_docs": {
+                "function": ingest_markdown_docs,
+                "description": "Ingest markdown documentation files with metadata extraction, batch processing, and progress tracking. Supports both single files and directories.",
```

Pylint (3.3.7)

[convention] Line too long (180/100)

(C0301)